scipy/stats/tests/test_qmc.py

```diff
@@ -168,48 +169,66 @@ def test_update_discrepancy(self):
         with pytest.raises(ValueError, match=r"x_new is not a 1D array"):
             update_discrepancy(x_new, space_1[:-1], disc_init)
 
+    def test_perm_discrepancy(self):
+        doe_init = np.array([[1, 3], [2, 6], [3, 2], [4, 5], [5, 1], [6, 4]])
+        doe_init = (2.0 * doe_init - 1.0) / (2.0 * 6.0)
+
+        disc_init = qmc.discrepancy(doe_init)
+
+        row_1, row_2, col = 5, 2, 1
+
+        doe = doe_init.copy()
+        doe[row_1, col], doe[row_2, col] = doe[row_2, col], doe[row_1, col]
+
+        disc_valid = qmc.discrepancy(doe)
+        disc_perm = _perturb_discrepancy(doe_init, row_1, row_2, col,
+                                         disc_init)
+
+        assert_allclose(disc_valid, disc_perm)
```

Comment:
Agreed I will update the tests.
Thanks for the analysis. It's indeed interesting to see the error increasing. I wonder what is causing this, some accumulation somewhere? As long as it stays small it's fine though.

Reply:
I guess I'm surprised that the error accumulates linearly; if it is not unstable, I would expect it to be more random, even at large scale. But I would be surprised if the equations are _wrong_ since it is so close